### PR TITLE
Authentication for developers

### PR DESCRIPTION
**Pull request by vikoivun**
## Description :sparkles:

Adds description of the tools used for authentication. Currently rather low level.

### CODE REVIEW

**Comment by vikoivun** [comment]
Now with introductory text!

File: docs/authentication.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Authentication
+
+Most of the projects requiring any sort of user authentication should use Tunnistamo.
+Tunnistamo is service that authenticates users against several different identity sources,
```
Comment (petterill) [inline docs/authentication.md]:
```suggestion
Tunnistamo is a service that authenticates users against several different identity sources,
```

File: docs/authentication.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+Tunnistamo also keeps a running session for the user, meaning the user won't be needing
+to login again to access another service using tunnistamo.
+
+The actual line protocols are OAuth2 and OIDC. New applications should always use OIDC.
+
+*Nomenclature alert*: Tunnistamo will soon be changing names, as the implementation will switch from your own implementation (named "tunnistamo") to Keycloak based implementation. Keeping the same name would be too confusing. We're not sure of the upcoming name yet.
+
+## Getting necessary Tunnistamo identifiers
+
+To get started you will either need a locally setup instance of Tunnistamo or get access to one of shared instances available within Helsinki infrastructure. I'd recommend that you try and get access to the shared instances, unless you must develop offline.
```
Comment (petterill) [inline docs/authentication.md]:
```suggestion
To get started you will either need a locally setup instance of Tunnistamo or get access to one of shared instances available within Helsinki infrastructure. It is recommended that you try and get access to the shared instances, unless you must develop offline. 
```

File: docs/authentication.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## Getting necessary Tunnistamo identifiers
+
+To get started you will either need a locally setup instance of Tunnistamo or get access to one of shared instances available within Helsinki infrastructure. I'd recommend that you try and get access to the shared instances, unless you must develop offline. 
+
+Current instances are:
+* Kanslia (executive office) instance: default instance used by projects, both during development and production
+* KuVa (culture and leisure) instance: instance used by KuVa projects during their development. Very flexible registration
+
+To get access to Kanslia instance, pop by the slack channel #tunnistamo
```
Comment (petterill) [inline docs/authentication.md]:
```suggestion
To get access to Kanslia instance, pop by the slack channel #tunnistamo.
```

**Review by petterill — COMMENTED** [review]
Looks good to me. Just some stylistic suggestions.

File: docs/authentication.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+# Authentication
```
Comment (petterill) [inline docs/authentication.md:7]:
```suggestion
---
name: Authentication
menu: Best practices
route: /authentication
---

# Authentication
```

**Review by petterill — CHANGES_REQUESTED** [review]
Forgot to add that the metadata must be added for the doc to show up on the site correctly

**Review by petterill — APPROVED** [review]

**Pull request merged**